Update the change log from v2.0.0 (#352)
nextcloud/integration_openproject: CHANGELOG.md

and this project adheres to [Semantic Versioning](http://semver.org/).

## [Unreleased]

> ## 2.1.0 – 2023-01-27

Reviewer: ```suggestion
## 2.2.1 – 2023-01-27
```

* use newly created branch of 2.0 for translations by @individual-it in [#237](https://github.com/nextcloud/integration_openproject/pull/237)
* run CI with stable25 branch of NC by @individual-it in [#239](https://github.com/nextcloud/integration_openproject/pull/239)
* do not translate oauth technical terms by @individual-it in [#241](https://github.com/nextcloud/integration_openproject/pull/241)
* bump version to 2.0.4 by @individual-it in [#242](https://github.com/nextcloud/integration_openproject/pull/242)
* build 2.0 using NC 24 by @individual-it in [#245](https://github.com/nextcloud/integration_openproject/pull/245)

## 2.0.3 – 2022-09-01
### Changed
* l10n: Separate words (backport of #210) by @individual-it in [#211](https://github.com/nextcloud/integration_openproject/pull/211)
* Updated description for the OpenProject app (backport to 2.0) by @individual-it in [#216](https://github.com/nextcloud/integration_openproject/pull/216)
* allow transifex to push to 2.0 branch by @individual-it in [#212](https://github.com/nextcloud/integration_openproject/pull/212)

> * OP#43843 fix app icon for notifications by @individual-it in [#218](https://github.com/nextcloud/integration_openproject/pull/218)

Reviewer: ```suggestion
* fix app icon for notifications by @individual-it in [#218](https://github.com/nextcloud/integration_openproject/pull/218)
```

* implement phpstan by @individual-it in [#39](https://github.com/nextcloud/integration_openproject/pull/39)
* set background image size to fit into the element background by @kiranparajuli589 in [#45](https://github.com/nextcloud/integration_openproject/pull/45)
* fix getOpenProjectAvatar error handling by @individual-it in [#47](https://github.com/nextcloud/integration_openproject/pull/47)
* Feature search workpackages by @SwikritiT in [#43](https://github.com/nextcloud/integration_openproject/pull/43)
* construct filters from array by @individual-it in [#48](https://github.com/nextcloud/integration_openproject/pull/48)
* Trigger CI for every push by @kiranparajuli589 in [#50]https://github.com/nextcloud/integration_openproject/pull/50
* Separate words by @Valdnet in [#54](https://github.com/nextcloud/integration_openproject/pull/54)
* cancel running workflows for the PR except the latest one by @kiranparajuli589 in [#52](https://github.com/nextcloud/integration_openproject/pull/52)
* run CI with multiple versions of PHP (7.4, 8.0) by @kiranparajuli589 in [#51](https://github.com/nextcloud/integration_openproject/pull/51)

> * OP#40534 only allow user to connect via OAuth by @individual-it in [#35](https://github.com/nextcloud/integration_openproject/pull/35)
> * OP#40534 delete anything to do with API tokens by @individual-it in [#46](https://github.com/nextcloud/integration_openproject/pull/46)

Reviewer: ```suggestion
* only allow user to connect via OAuth by @individual-it in [#35](https://github.com/nextcloud/integration_openproject/pull/35)
* delete anything to do with API tokens by @individual-it in [#46](https://github.com/nextcloud/integration_openproject/pull/46)
```

* the search result isnt cleared when a work package is selected by @kiranparajuli589 in [#98](https://github.com/nextcloud/integration_openproject/pull/98)
* Add a seperator between worpackages search list by @SwikritiT in [#101](https://github.com/nextcloud/integration_openproject/pull/101)
* add rules related to stylelint spaces and newlines by @kiranparajuli589 in [#103](https://github.com/nextcloud/integration_openproject/pull/103)
* Add new OCS endpoint to get file info from file ID by @julien-nc in [#67](https://github.com/nextcloud/integration_openproject/pull/67)
* use states for every components from a single global source by @kiranparajuli589 in [#100](https://github.com/nextcloud/integration_openproject/pull/100)
* only test with NC 23 by @individual-it in [#105](https://github.com/nextcloud/integration_openproject/pull/105)
* relations/notifications even when the oauth information is not complete by @kiranparajuli589 in [#99](https://github.com/nextcloud/integration_openproject/pull/99)
* run more test variations in CI by @individual-it in [#106](https://github.com/nextcloud/integration_openproject/pull/106)

> * OP#41271 remember new refresh_token after refreshing oauth token by @individual-it in [#108](https://github.com/nextcloud/integration_openproject/pull/108)
> * OP#41949 allways send storageUrl by @individual-it in [#104](https://github.com/nextcloud/integration_openproject/pull/104)

Reviewer: ```suggestion
* remember new refresh_token after refreshing oauth token by @individual-it in [#108](https://github.com/nextcloud/integration_openproject/pull/108)
*  always send storageUrl by @individual-it in [#104](https://github.com/nextcloud/integration_openproject/pull/104)
```

* highlight & focus the OP instance input on invalid input by @individual-it in [#123](https://github.com/nextcloud/integration_openproject/pull/123)
* Automatically create a NC OAuth client for OpenProject by @julien-nc in [#97](https://github.com/nextcloud/integration_openproject/pull/97)
* Fix unit tests to not display ugly log by @SwikritiT in [#127](https://github.com/nextcloud/integration_openproject/pull/127)
* Use a dedicated section for admin settings by @julien-nc in [#128](https://github.com/nextcloud/integration_openproject/pull/128)
* Don't show connect button if the server is correctly set-up but offline by @SwikritiT in [#130](https://github.com/nextcloud/integration_openproject/pull/130)
* Fix problem with search list not clearing by @SwikritiT in [#134](https://github.com/nextcloud/integration_openproject/pull/134)
* New oauth service method to set the redirect uri by @julien-nc in [#136](https://github.com/nextcloud/integration_openproject/pull/136)
* Sticky search input by @kiranparajuli589 in [#135](https://github.com/nextcloud/integration_openproject/pull/135)

> * OP#42613 cache colors of statuses and types by @individual-it in [#129](https://github.com/nextcloud/integration_openproject/pull/129)

Reviewer: ```suggestion
* cache colors of statuses and types by @individual-it in [#129](https://github.com/nextcloud/integration_openproject/pull/129)
```

> * OP#42613 cache colors of types and statuses in the frontedn by @individual-it in [#137](https://github.com/nextcloud/integration_openproject/pull/137)

Reviewer: ```suggestion
*  cache colors of types and statuses in the frontedn by @individual-it in [#137](https://github.com/nextcloud/integration_openproject/pull/137)
```

* House keeping :grin: by @julien-nc in [#145](https://github.com/nextcloud/integration_openproject/pull/145)
* Use toggle icon for the encrypted field value inspection by @kiranparajuli589 in [#141](https://github.com/nextcloud/integration_openproject/pull/141)
* only show copy button when copy to clipboard is possible by @individual-it in [#149](https://github.com/nextcloud/integration_openproject/pull/149)
* l10n: Delete dot and change to a lowercase letter by @Valdnet in [#150](https://github.com/nextcloud/integration_openproject/pull/150)
* redirect user back to where oauth journey was started by @individual-it in [#139](https://github.com/nextcloud/integration_openproject/pull/139)
* UI/UX improvements: use grey color for readonly input fields by @kiranparajuli589 in [#151](https://github.com/nextcloud/integration_openproject/pull/151)
* Remove requestUrl from front-end by @SwikritiT in [#152](https://github.com/nextcloud/integration_openproject/pull/152)
* Improve developers documentation by @SwikritiT in [#132](https://github.com/nextcloud/integration_openproject/pull/132)

> * OP#42726 manual testplan by @individual-it in [#133](https://github.com/nextcloud/integration_openproject/pull/133)

Reviewer: ```suggestion
* manual testplan by @individual-it in [#133](https://github.com/nextcloud/integration_openproject/pull/133)
```

* pass in the OpenProject URL as state to the ProjectsTab by @individual-it in [#259](https://github.com/nextcloud/integration_openproject/pull/259)
* remove the hard-coded color values or the search bar component by @kiranparajuli589 in [#263](https://github.com/nextcloud/integration_openproject/pull/263)
* fix fetching avatars from openproject by @individual-it in [#313](https://github.com/nextcloud/integration_openproject/pull/313)
* set defaults for oauth connection message&result by @individual-it in [#312](https://github.com/nextcloud/integration_openproject/pull/312)
* Fix inconsistencies in dark theme - release/2.2 by @SwikritiT in [#327](https://github.com/nextcloud/integration_openproject/pull/327)
* Fix the background colour for search bar by @SwikritiT in [#281](https://github.com/nextcloud/integration_openproject/pull/281)
* Fix revocation while the default settings changed by @SwikritiT in [#291](https://github.com/nextcloud/integration_openproject/pull/291)
* Fix search thumbnail by @SwikritiT in [#333](https://github.com/nextcloud/integration_openproject/pull/333)

> 

Reviewer: ```suggestion
```

* move sharing related tests to new context by @individual-it in [#294](https://github.com/nextcloud/integration_openproject/pull/294)
* make reason strings translatable by @individual-it in [#295](https://github.com/nextcloud/integration_openproject/pull/295)
* mention all notification reasons by @individual-it in [#307](https://github.com/nextcloud/integration_openproject/pull/307)
* Update dialog version by @SwikritiT in [#340](https://github.com/nextcloud/integration_openproject/pull/340)
* use php 8.1 for release by @individual-it in [#344](https://github.com/nextcloud/integration_openproject/pull/344)
* bump version to 2.2.1 by @individual-it in [#345](https://github.com/nextcloud/integration_openproject/pull/345)

### Removed
* stop showing OP notifications as NC notifications by @individual-it in [#282](https://github.com/nextcloud/integration_openproject/pull/282)
* Remove leftover of notification code by @SwikritiT in [#283](https://github.com/nextcloud/integration_openproject/pull/283)
* NC master is not compatible with PHP 7 by @individual-it in [#339](https://github.com/nextcloud/integration_openproject/pull/339)

> 
> 

Reviewer: ```suggestion

```

* Simplify css and fix icon style for NC 25 by @julien-nc in [#234](https://github.com/nextcloud/integration_openproject/pull/234)
* improve readme list with some grammer fixes by @kiranparajuli589 in [#240](https://github.com/nextcloud/integration_openproject/pull/240)
* show admin why connection to OpenProject cannot be established by @individual-it in [#217](https://github.com/nextcloud/integration_openproject/pull/217)
* Revert show admin why connection to OpenProject cannot be established" by @individual-it in [#243](https://github.com/nextcloud/integration_openproject/pull/243)
* merge release/2.0 changes into master by @individual-it in [#236](https://github.com/nextcloud/integration_openproject/pull/236)
* Bump max NC version to 26, adjust CI action by @julien-nc in [#246](https://github.com/nextcloud/integration_openproject/pull/246)
* show admin why connection to OpenProject cannot be established by @individual-it in [#244](https://github.com/nextcloud/integration_openproject/pull/244)
* bump version to 2.1 by @individual-it in [#250](https://github.com/nextcloud/integration_openproject/pull/250)

> 

Reviewer: ```suggestion
```

* Handle error while unlinking work packages by @SwikritiT in [#120](https://github.com/nextcloud/integration_openproject/pull/120)
* Fix icon placements and colors for different themes by @kiranparajuli589 in [#107](https://github.com/nextcloud/integration_openproject/pull/107)
* Don't show connect button if the server is correctly set-up but offline by @SwikritiT in [#130](https://github.com/nextcloud/integration_openproject/pull/130)
* do not use speaker icon for the notification items in the dashboard widget by @kiranparajuli589 in [#122](https://github.com/nextcloud/integration_openproject/pull/122)
* Fix problem with search list not clearing by @SwikritiT in [#134](https://github.com/nextcloud/integration_openproject/pull/134)
* reset OP host validity check by @individual-it in [#177](https://github.com/nextcloud/integration_openproject/pull/177)
* calculate contrast for wp type and show border if low by @individual-it in [#176](https://github.com/nextcloud/integration_openproject/pull/176)
* Unlink icon button is dark in the dark mode by @kiranparajuli589 in [#185](https://github.com/nextcloud/integration_openproject/pull/185)

> 

Reviewer: ```suggestion
```
